### DISCUSSION TITLE
How do you create more than one VM from the same ISO?

### DISCUSSION THREAD

**Discussion by nclsbayona** [quickemu-project/quickemu#839] [Quickemu] [answered]
Hi. This is a question / feature request, with that I mean I want to create more than one VM from the same ISO and I can't find a way to do so, let's say for example I already downloaded Ubuntu Mate 22.04 ISO but now I want to create not only one VM but two or more, the problem I'm facing now is that just one .conf file is created and even if I copy that file again, they would be pointing to the same .qcow file and I don't know if there's a way to create one using all of quickemu's advantages. With that being said, I think that It would be really great if quickemu uses a directory structure to organize VMs, allowing an option to create another VM out of an ISO

**Comment by nclsbayona** [comment]
Quick edit. I also found that even if you create two different qcow files, the IP assigned to each VM is the same 
![image](https://user-images.githubusercontent.com/59931437/215893832-4c3abc04-ce0f-4e83-8209-5cf28fb57995.png)

**Comment by philclifford** [comment, marked as answer]
You really need a separate .conf file pointing to a separate directory for each VM (to keep the log files, pid file, port assignment records and virtual disks together and avoid getting in a mess). Just create the first/template using `quickget` and then make new directories and config files pointing to them with a little editing.  The simple thing (to my simple mind) is to link the fetched ISO into the new directories, adjust the paths in the new config and let `quickemu` take care of the rest. Something like :

``` bash
for clone in 1 2 3 ; do   
  mkdir ubuntu-mate-24.04-${clone} &&   \
  ln ubuntu-mate-24.04/*.iso ubuntu-mate-24.04-${clone}/ && \
  sed "s/ubuntu-mate-24.04\//ubuntu-mate-24.04-${clone}\//"  \
       ubuntu-mate-24.04.conf  >ubuntu-mate-24.04-${clone}.conf
  chmod --reference ubuntu-mate-24.04.conf ubuntu-mate-24.04-${clone}.conf
done
```

That will skip directories that already exist but should end up with something like this:

![image](https://github.com/quickemu-project/quickemu/assets/8797027/08cf3e91-43d4-452d-8a79-3dd6e8811119)

where only the original and "clone-2" have been started and all the installation isos are hard links so reference "the same file".
You can tweak the .conf files further (for example each will sort of get half the free memory by default as you start them up so you might want to assign them all equal memory, and you might want to pin down their ssh port mappings so that they are not assigned from the default range - I usually stick them outside the range that `quickemu` uses and put the host names and ports into my .ssh/config). e.g.

``` text
Host qvm-kdeneon
port 24225
Host qvm-zorin
port 24228
Host qvm-popos
port 24229
```
You can give the config files/directories more meaningful names if you wish: they will be reflected in the VM name (and default hostname).
